cgp/genome.py

```diff
@@ -270,6 +269,105 @@ def randomize(self, rng: np.random.RandomState) -> None:
         # accept generated dna if it is valid
         self.dna = dna
 
+    def splice_dna(self, dna_insert: List[int], internal_start_node: int = 0) -> int:
+        """Splice the dna.
+
+        Replaces a segment of the dna starting from internal start node with the dna insert
+
+        Parameters
+        ----------
+        dna_insert: List[int]
+            dna segment to be inserted.
+        internal_start_node: int
+            idx of the internal node, where the insert starts.
+            Defaults to 0.
+
+        Returns
+        ----------
+        address_last_inserted_node: int
+            address of the last node inserted.
+        """
+        if len(dna_insert) % self._length_per_region != 0 or len(dna_insert) == 0:
+            raise ValueError("Length of inserted dna must be a multiple of region length")
+
+        n_inserted_nodes = len(dna_insert) // self._length_per_region
+
+        if internal_start_node + n_inserted_nodes >= self._n_hidden:
+            raise ValueError("Inserted dna too long")
+
+        dna = self.dna
+
+        start_idx = (self._n_inputs + internal_start_node) * self._length_per_region
+        end_idx = start_idx + len(dna_insert)
+        dna[start_idx:end_idx] = dna_insert
+
+        self.dna = dna
+
+        address_last_inserted_node = self._n_inputs + internal_start_node + n_inserted_nodes - 1
```

Comment:
the "address" is just the same as the index, right? so we could drop the "address_" prefix here, it doesn't provide more information. we also say "internal_start_node" and not "internal_start_node_address"